blaze/compute/sql.py

```diff
@@ -842,6 +842,24 @@ def compute_up(t, s, **kwargs):
     return s.order_by(*cols)
 
 
+@dispatch(Sample, sa.Table)
+def compute_up(t, s, **kwargs):
+    n = t.n if t.n is not None else int(t.frac * s.count().scalar())
```

Comment:
Yes, there's a lot of room here for performance improvements.  I'll look into the subselect.  Related questions:  is sqla sophisticated enough to translate `s.count()` into an O(1) operation if the underlying table is indexed?  Is this dialect / engine specific?  Or does sqla take the safe approach an always do a table scan regardless?